pyuvdata/uvdata/initializers.py

```diff
@@ -0,0 +1,564 @@
+    phase_center_catalog: dict[str, Any] | None = None,
+    phase_center_id_array: np.ndarray | None = None,
+    **kwargs,
+):
+    """Initialize a new UVData object from keyword arguments.
+
+    Parameters
+    ----------
+    freq_array : ndarray of float
+        Array of frequencies in Hz.
+    polarization_array : sequence of int or str
+        Array of polarization integers or strings (eg. 'xx' or 'ee')
+    antenna_positions : ndarray of float or dict of ndarray of float
+        Array of antenna positions in ECEF coordinates in meters.
+        If a dict, keys are antenna names or numbers and values are
+        antenna positions in ECEF coordinates in meters.
+    telescope_location : astropy EarthLocation or MoonLocation
+        Location of the telescope.
+    telescope_name : str
+        Name of the telescope.
+    unique_times : ndarray of float, optional
+        Array of unique times in Julian Date. If provided, unique_antpairs must also
+        be provided, and the blt axis of the UVData object will be formed from the
+        outer product of unique_times and unique_antpairs. If not provided, time_array
+        must be provided.
+    time_array : ndarray of float, optional
+        Array of times in Julian Date. If provided, antpair_array must also be
+        provided. Required if unique_times is not provided. If time_array is provided,
+        it will be passed on as-is to the UVData object.
+    unique_antpairs : sequence of 2-tuple of int or 2D array of int, optional
+        Sequence of unique antenna pairs in the data. If provided, unique_times must
+        also be provided. If not, antpair_array must be provided. If provided, the blt
+        axis of the UVData object will be formed from the outer product of unique_times
+        and unique_antpairs.
+    antpair_array : sequence of 2-tuples of int or 2D array of int, optional
+        Antenna pairs in the data. If an ndarray, must have shape (Nants, 2).
+        If provided, time_array must also be
+        provided. Required if unique_antpairs is not provided. If antpair_array is
+        provided, it will be converted to baseline_array in the same order as provided,
+        and this is provided to the UVData object.
+    integration_time : float or ndarray of float, optional
+        Integration time in seconds. If not provided, it will be derived from the
+        time_array, as the difference between successive times (with the last time
+        appended). If a float is provided, it will be used for all integrations.
+        If an ndarray is provided, it must have the same shape as time_array (or
+        unique_times, if that is what is provided).
+    channel_width : float or ndarray of float, optional
+        Channel width in Hz. If not provided, it will be derived from the freq_array,
+        as the difference between successive frequencies (with the last frequency
+        appended). If a float is provided, it will be used for all channels.
```

Comment:
This isn't quite right -- the last frequency isn't appended, the last frequency difference is appended, right?